### PR TITLE
Some automated code quality and code style fixes created by Drupal8 Rector

### PR DESCRIPTION
**Pull request by mxr576**
Related commits in this PR

1. 68220c1 - Code optimization
2. bf0a6f6 - Fixing inconsistent variable/property names according to Drupal coding standard.
3. c1e5a4c - Fixes by a new rector that solves a regular issue caused by Drupal 8 traits.

(Because this PR already contains fixes for #138)

Started to work on a tool that can apply automated fixes related to general- and Drupal 8 specific code quality and code style issues that neither PHPCS nor any other existing tools can do.

This a PR with the issues that the very first version of this tool has found and fixed. Fixing inconsistent property/variable names just a small thing compared with the automated condition optimization, dead-code removal and replacing deprecated method calls with the right ones.

I would not add this tool just yet to our automated Travis builds because it is highly WIP state, but the issues that it fixed should be applied now.

Link to my library: https://packagist.org/packages/mxr576/drupal8-rector 

Latest Travis build: https://travis-ci.org/mxr576/apigee-devportal-drupal/builds/497059264

### CODE REVIEW

File: modules/apigee_edge_debug/src/HttpClientMiddleware/ApiClientProfiler.php [hunk 1]
```diff
@@ -55,13 +55,6 @@ class ApiClientProfiler {
    */
   private $logFormat;
 
-  /**
-   * The debug message formatter plugin manager.
-   *
-   * @var \Drupal\apigee_edge_debug\DebugMessageFormatterPluginManager
-   */
-  private $debugMessageFormatterPlugin;
```
Comment (mxr576) [inline modules/apigee_edge_debug/src/HttpClientMiddleware/ApiClientProfiler.php:63]: Dead code detected.

File: src/Entity/FieldableEdgeEntityBase.php [hunk 2]
```diff
@@ -312,7 +312,7 @@ public function getFieldDefinition($name) {
    */
   public function getFieldDefinitions() {
     if (!isset($this->fieldDefinitions)) {
-      $this->fieldDefinitions = $this->entityManager()->getFieldDefinitions($this->entityTypeId, $this->bundle());
```
Comment (mxr576) [inline src/Entity/FieldableEdgeEntityBase.php:315]: Fixed usage of a deprecated method.

File: src/Controller/ExportAnalyticsController.php [hunk 3]
```diff
@@ -86,11 +86,13 @@ public function exportAsCsv($data_id): Response {
     $data = [];
 
     // Write out the data.
-    for ($i = 0; $i < count($analytics['TimeUnit']); $i++) {
-      $data[] = [
-        $this->t('Date')->render() => new DrupalDateTime('@' . $analytics['TimeUnit'][$i] / 1000),
-        $analytics['metric']->render() => $analytics['stats']['data'][0]['metric'][0]['values'][$i],
-      ];
+    if (array_key_exists('TimeUnit', $analytics) && is_array($analytics['TimeUnit'])) {
```
Comment (mxr576) [inline src/Controller/ExportAnalyticsController.php:89]: Make sure we do not want to count an uncountable. (The original automated fix checked whether  `$analytics['TimeUnit']` is an array or countable. I optimized this fix manually.)

File: src/Entity/Storage/DeveloperStorage.php [hunk 4]
```diff
@@ -119,15 +119,9 @@ protected function entityController(): EdgeEntityControllerInterface {
...
-        // It could be an integer if developer UUID has been used as as an id
-        // instead of the email address.
-        if (is_object($entity)) {
-          /** @var \Drupal\apigee_edge\Entity\DeveloperInterface $entity */
-          $entitiesByDeveloperId[$entity->getDeveloperId()] = $entity;
-        }
-      }
+      // It could be an integer if developer UUID has been used as as an id
+      // instead of the email address.
+      $entitiesByDeveloperId = array_filter($entities, 'is_object');
```
Comment (mxr576) [inline src/Entity/Storage/DeveloperStorage.php]: Replaced `foreach() `with `array_filter()` that does not spam the method's scope with remnant variables.

File: src/Entity/Storage/EdgeEntityStorageBase.php [hunk 5]
```diff
@@ -254,7 +256,7 @@ protected function getFromStorage(array $ids = NULL) {
     // If ids is an empty array there is nothing to do.
     // Probably every entities could have been found in the persistent cache.
     // Node::loadMultiple() works the same.
-    if (is_array($ids) && empty($ids)) {
+    if ($ids === []) {
```
Comment (mxr576) [inline src/Entity/Storage/EdgeEntityStorageBase.php:259]: Optimized type check.

**Review by Jaesin — APPROVED** [review]
Tests for `apigee_m10n` pass. this looks good to me.

**Comment by mxr576** [comment]
Rebased after #138 got merged.

**Pull request merged**